docs/content/en/getting-started/module-quick-start.md

```diff
@@ -0,0 +1,223 @@
+
+The most important directory is the `templates/` directory, which will contain any templates we want to render.
+
+## Step 2: Creating a Template
+
+Let's create a template that creates a simple hello world message in Go. We'll start by creating a `hello.go.tpl` in the `templates/` directory.
+
+{{< code file="hello.go.tpl" copy=true >}}
+package main
+
+func main() {
+	fmt.Println("Hello, world!")
+}
+{{< /code >}}
+
+## Step 3: Consuming the Module in an Application
+
+Now that we've done that, well how do we use it? This is super easy with the `replacements` map in a [`service.yaml`](/stencil/reference/service.yaml). 
+
+Let's quickly create a test application:
+
+{{< code file="create-test-module.sh" copy=true >}}
+mkdir testapp; cd testapp
+cat > service.yaml <<EOF
+name: testapp
+modules:
+- name: github.com/yourorg/helloworld
+replacements:
+	# Replace ../helloworld with the path to your module.
+	github.com/yourorg/helloworld: ../helloworld
+{{< /code >}}
+
+Now if we run stencil on the test application, we should see the following:
+
+```bash
+testapp ❯ stencil
+INFO[0000] stencil v1.14.2
+INFO[0000] Fetching dependencies
+INFO[0002]  -> github.com/yourorg/helloworld local
+INFO[0002] Loading native extensions
+INFO[0002] Rendering templates
+INFO[0002] Writing template(s) to disk
+INFO[0002]   -> Created hello.go
+```
+
+It looks like it created `hello.go` for us! Let's validate:
+
+```bash
+testapp ❯ cat hello.go
+package main
+
+func main() {
+	fmt.Println("Hello, world!")
+}
+```
+
+:tada: We have a hello world application!
+
+## Step 4: Using a Block
+
+Blocks are incredibly easy to use in Stencil.
+
+{{% note %}}
+In case you don't remember, blocks are areas in your generated code that you'd like to persist across runs.
+{{% /note %}}
+
+Let's create our own block in the hello.go template from earlier:
+
+{{< code file="hello.go.tpl" copy=true >}}
+package main
+
+func main() {
+	fmt.Println("Hello, world!")
+	###Block(additionalMessage)
+{{- /* It's important to not indent the file.Block to prevent the indentation from being copied over and.. over again. */ }}
+{{ file.Block "additionalMessage" }}
+	###EndBlock(additionalMessage)
+}
+{{< /code >}}
+
+If we re-run stencil and look at `hello.go` we should see the following:
+
+
+{{< code file="hello.go" copy=true >}}
+package main
+
+func main() {
+	fmt.Println("Hello, world!")
+	###Block(additionalMessage)
+
+	###EndBlock(additionalMessage)
+}
+{{< /code >}}
+
+If we add contents to the block and re-run stencil they'll be persisted across the run!
+
+## Step 5: (Optional) Creating Multiple Files
+
+One of the powerful parts of stencil is the ability to create an arbitrary number of files with a single template. This is done with the [`file.Create`](/stencil/functions/file.create) function. Let's create a `greeter.go.tpl` template in the `templates/` directory that'll create a `hello.go` and `goodbye.go` file.
```

Comment:
Command line tools are a good example. We create multiple clis based on the `commands: []` parameter in [stencil-golang (ref)](https://github.com/getoutreach/stencil-golang/blob/main/templates/cmd/main_cli.go.tpl). So, we need a single template that's able to create multiple files on disk.

It's actually probably one of the most useful parts of stencil and is going to become even more apparent when I expose this functionality to native extensions in the future so that they themselves can also write files.

Reply:
oic.  i'd use something like that as the example then -- this example makes no sense because i don't see why it shouldn't just be two template files.  if you had this creating files based on looping through an input list it would be immediately clear why it exists.